sciencebeam_gym/preprocess/find_file_pairs_test.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import os
+from mock import patch
+
+import pytest
+
+import sciencebeam_gym.preprocess.find_file_pairs as find_file_pairs
+from sciencebeam_gym.preprocess.find_file_pairs import (
+  run,
+  parse_args,
+  main
+)
+
+
+LOGGER = logging.getLogger(__name__)
+
+BASE_SOURCE_PATH = '/source'
+
+PDF_FILE_1 = BASE_SOURCE_PATH + '/file1.pdf'
+XML_FILE_1 = BASE_SOURCE_PATH + '/file1.xml'
+PDF_FILE_2 = BASE_SOURCE_PATH + '/file2.pdf'
+XML_FILE_2 = BASE_SOURCE_PATH + '/file2.xml'
+
+SOURCE_PATTERN = '*.pdf'
+XML_PATTERN = '*.xml'
+OUTPUT_FILE = 'file-list.tsv'
+
+SOME_ARGV = [
+  '--data-path=%s' % BASE_SOURCE_PATH,
+  '--source-pattern=%s' % SOURCE_PATTERN,
+  '--xml-pattern=%s' % XML_PATTERN,
+  '--out=%s' % OUTPUT_FILE
+]
+
+
+@pytest.fixture(name='find_file_pairs_grouped_by_parent_directory_or_name_mock')
+def _find_file_pairs_grouped_by_parent_directory_or_name():
+  with patch.object(find_file_pairs, 'find_file_pairs_grouped_by_parent_directory_or_name') as m:
+    yield m
+
+@pytest.fixture(name='save_file_pairs_to_csv_mock')
+def _save_file_pairs_to_csv():
+  with patch.object(find_file_pairs, 'save_file_pairs_to_csv') as m:
+    yield m
+
+@pytest.fixture(name='save_file_pairs_to_csv_mock')
+def _save_file_pairs_to_csv():
+  with patch.object(find_file_pairs, 'save_file_pairs_to_csv') as m:
+    yield m
+
+@pytest.fixture(name='parse_args_mock')
+def _parse_args():
+  with patch.object(find_file_pairs, 'parse_args') as m:
+    yield m
+
+@pytest.fixture(name='run_mock')
+def _run():
+  with patch.object(find_file_pairs, 'run') as m:
+    yield m
+
+class TestRun(object):
+  def test_should_pass_around_parameters(
+    self,
+    find_file_pairs_grouped_by_parent_directory_or_name_mock,
+    save_file_pairs_to_csv_mock):
+
+    opt = parse_args(SOME_ARGV)
+    find_file_pairs_grouped_by_parent_directory_or_name_mock.return_value = [
+      (PDF_FILE_1, XML_FILE_1),
+      (PDF_FILE_2, XML_FILE_2)
+    ]
+    run(opt)
+    find_file_pairs_grouped_by_parent_directory_or_name_mock.assert_called_with([
+      os.path.join(BASE_SOURCE_PATH, SOURCE_PATTERN),
+      os.path.join(BASE_SOURCE_PATH, XML_PATTERN)
+    ])
+    save_file_pairs_to_csv_mock.assert_called_with(
+      opt.out,
+      find_file_pairs_grouped_by_parent_directory_or_name_mock.return_value
+    )
+
+  def test_should_generate_file_list(self, tmpdir):
```

Comment:
Maybe some of these prep steps could be combined in a fixture to reduce the test case code? not a blocker just a thought.

Reply:
I've updated it.